numpy/ma/core.py

```diff
@@ -1481,9 +1481,14 @@ def getmaskarray(arr):
     """
     mask = getmask(arr)
     if mask is nomask:
-        mask = make_mask_none(np.shape(arr), getattr(arr, 'dtype', None))
+        mask = make_mask_none(
+            np.shape(arr),
+            dtype=getattr(arr, 'dtype', None),
+            readonly=nomask_as_readonly)
     return mask
 
+def _viewmaskarray(arr):
```

Comment:
That seems to expose unecessary implementation details. The important thing is whether the consumer is ok with `mask.writeable == False`. Everything else is just an optimization.